app/scripts/migrations/144.ts

```diff
@@ -0,0 +1,71 @@
+import { hasProperty } from '@metamask/utils';
+import { cloneDeep, isObject } from 'lodash';
+import availableCurrencies from '../../../ui/helpers/constants/available-conversions.json';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+type CurrencyController = {
+  currentCurrency?: string;
+};
+
+export const version = 144;
+const DEFAULT_CURRENCY = 'usd';
+
+/**
+ * This migration ensures that the `currentCurrency` in `CurrencyController`
+ * is set to a valid available currency. If it's missing or invalid, it defaults to "USD".
+ *
+ * @param originalVersionedData - The original MetaMask extension state.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (!hasProperty(state, 'CurrencyController')) {
+    global.sentry?.captureException?.(
+      new Error(`Migration ${version}: Missing CurrencyController in state`),
+    );
+    return;
+  }
+
+  const currencyController = state.CurrencyController as CurrencyController;
+
+  if (!isObject(currencyController)) {
+    global.sentry?.captureException?.(
+      new Error(
+        `Migration ${version}: Invalid CurrencyController state type '${typeof currencyController}'`,
+      ),
+    );
+    return;
+  }
+
+  const currentCurrency = currencyController.currentCurrency;
+
+  if (!currentCurrency) {
+    global.sentry?.captureException?.(
+      new Error(
+        `Migration ${version}: Missing currentCurrency in CurrencyController`,
+      ),
+    );
```

Comment:
Hmm should we give more context to this error? Something like "missing currency, defaulting to ${defaultCurrency}"?

Reply:
Should we produce an error?

Reply:
done [92b38df](https://github.com/MetaMask/metamask-extension/pull/30324/commits/92b38dfdca4e929683607002d5a77772a83372ea)